server/src/test/java/org/spine3/server/projection/ProjectionRepositoryShould.java

```diff
@@ -249,6 +266,48 @@ private void ensureCatchesUpFromEventStorage(ProjectionRepository<ProjectId, Tes
         assertTrue(TestProjection.processed(Events.getMessage(projectStartedEvent)));
     }
 
+    @Test
+    public void use_id_set_function() {
+        final IdSetFunction<ProjectId, ProjectCreated, EventContext> delegateFn =
+                new IdSetFunction<ProjectId, ProjectCreated, EventContext>() {
+                    @Override
+                    public Set<ProjectId> apply(ProjectCreated message, EventContext context) {
+                        return newHashSet();
+                    }
+                };
+
+        final IdSetFunction<ProjectId, ProjectCreated, EventContext> idSetFunction = spy(delegateFn);
+        repository.addIdSetFunction(ProjectCreated.class, idSetFunction);
+
+        final Event event = Given.Event.projectCreated(ID);
+        repository.dispatch(event);
+
+        final ProjectCreated expectedEventMessage = Events.getMessage(event);
+        final EventContext context = event.getContext();
+        verify(idSetFunction).apply(eq(expectedEventMessage), eq(context));
+    }
+
+    @Test
+    public void obtain_id_set_function_after_put() {
+        repository.addIdSetFunction(ProjectCreated.class, idSetForCreateProject);
+
+        final Optional<IdSetFunction<ProjectId, ProjectCreated, EventContext>> out = repository.getIdSetFunction(ProjectCreated.class);
+
+        assertTrue(out.isPresent());
+        //noinspection OptionalGetWithoutIsPresent
```

Comment:
Please use an annotation-based suppression instead. It's OK, if it is for the whole method, as this is a test codebase.